craft_application/commands/init.py

```diff
@@ -101,8 +102,19 @@ def profiles(self) -> list[str]:
 
     def run(self, parsed_args: argparse.Namespace) -> None:
         """Run the command."""
-        project_dir = self._get_project_dir(parsed_args)
+        # If the user provided a "name" and it's not valid, the command fails.
+        if parsed_args.name is not None:
+            self._services.init.validate_project_name(parsed_args.name)
+
+        # However, if the name comes from the directory, we don't fail and
+        # instead fallback to "my-project".
         project_name = self._get_name(parsed_args)
+        try:
+            self._services.init.validate_project_name(project_name)
+        except errors.InitError:
+            project_name = "my-project"
```

Comment:
What about moving this logic to `self._services.init.validate_project_name(project_name, fallback="my-project")` and if you specify `fallback=` it will return it instead of raising an exception (like `.get()` when you specify `default` )?

I'm also thinking if it would allow to store `my-project` as a `InitService` property, that you may override in you specified service

Reply:
Hmm I like it; how about this signature then?
`validate_project_name(name: str, use_default: bool = False)`
... and then if `use_default` is True it returns `InitService.default_name`

Reply:
I ended up using a constructor parameter so that downstream applications can just use the `Application._configure_services()` that they likely already have, instead of having to write a "whole subclass" just to set an attribute - please take a look.

Reply:
This is the way